src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -445,12 +445,12 @@ static void window_new_ride_scroll_to_focused_ride(rct_window* w)
     // Find row index of the focused ride type
     rct_widget* listWidget = &window_new_ride_widgets[WIDX_RIDE_LIST];
     assert(_windowNewRideCurrentTab < std::size(_windowNewRideHighlightedItem));
-    int32_t focusRideType = _windowNewRideHighlightedItem[_windowNewRideCurrentTab].ride_type_and_entry;
+    auto focusRideType = _windowNewRideHighlightedItem[_windowNewRideCurrentTab];
     int32_t count = 0, row = 0;
-    ride_list_item* listItem = _windowNewRideListItems;
-    while (listItem->type != RIDE_TYPE_NULL || listItem->entry_index != RIDE_ENTRY_INDEX_NULL)
+    RideSelection* listItem = _windowNewRideListItems;
+    while (listItem->Type != RIDE_TYPE_NULL || listItem->EntryIndex != RIDE_ENTRY_INDEX_NULL)
     {
-        if (listItem->type == focusRideType)
+        if (listItem->Type == focusRideType.Type && listItem->EntryIndex == 0)
```

Comment:
This is what was done before. It could be that 

```suggestion
        if (listItem->Type == focusRideType.Type)
```
was intended.